src/engine/app.rs

```diff
@@ -56,20 +56,22 @@ impl Application {
 
     /// Advances the game world by one tick.
     fn advance_frame(&mut self) {
-        // let context = self.context.lock().unwrap().deref_mut();
-        let engine_events = self.context.lock().unwrap().poll_engine_events();
-        for engine_event in engine_events {
-            self.context.lock().unwrap().broadcaster.publish().with::<EngineEvent>(engine_event);
+        let events = self.context.lock().unwrap().poll_engine_events();
+        for e in events {
+            self.context.lock().unwrap().broadcaster.publish().with::<EngineEvent>(e);
         }
-        let events = self.context.lock().unwrap().broadcaster.poll();
-        self.states.handle_events(events, self.context.lock().unwrap().deref_mut());
+
+        let ents = self.context.lock().unwrap().broadcaster.poll();
```

Comment:
No, that was intentional. It's short for `entities`. For some reason, `broadcaster.poll()` returns a list of entities, not events, despite being passed into `handle_events()`. I don't think I reviewed the code when it was first committed, so I think a refactor of Broadcaster might be necessary.